.circleci/config.yml

```diff
@@ -406,6 +448,36 @@ jobs:
             - repo/scripts/implementation_coverage_aggregated.csv
             - repo/scripts/implementation_coverage_full.csv
 
+  generate-random-aws-credentials:
+    executor: ubuntu-machine-amd64
+    working_directory: /tmp/workspace/repo
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Generate Random AWS Account ID
+          command: |
+            # Generate a random 12-digit non-default account number
+            while true; do
+              # Generate a random 12-digit number for TEST_AWS_ACCOUNT_ID
+              export TEST_AWS_ACCOUNT_ID=$(LC_ALL=C tr -dc '0-9' < /dev/urandom | fold -w 12 | head -n 1)
+              # Check if the generated number is not 000000000000
+              if [ "$TEST_AWS_ACCOUNT_ID" != "000000000000" ]; then
+                break
+              fi
+            done
```

Comment:
nit: I'm not sure if it's really necessary to check that the randomly generated 12-digit number is not all-zeros (the default).  The chance is one in a trillion, right (assuming it's perfectly random)?